### PR TITLE
Fixes Preferences user file search order (#891)

### PR DESCRIPTION
**Pull request by MatthiasWM**
Tried to write really clean code...

### CODE REVIEW

File: src/drivers/Unix/Fl_Unix_System_Driver.cxx [hunk 1]
```diff
@@ -443,87 +444,135 @@ void Fl_Unix_System_Driver::newUUID(char *uuidBuffer)
...
+      prefs_path_14.replace(h_env, 5, home_path);
+  }
+  if (prefs_path_14[prefs_path_14.size()-1]!='/')
+    prefs_path_14.append('/');
+  prefs_path_14.append(vendor);
+
+  // 2: If this base path does not exist, try the 1.3 path
+  if (::access(prefs_path_14.c_str(), F_OK) == -1) {
+    Fl_String prefs_path_13 = home_path + ".fltk/" + vendor;
+    if (::access(prefs_path_14.c_str(), F_OK) == 0) {
```
Comment (Albrecht-S) [inline src/drivers/Unix/Fl_Unix_System_Driver.cxx:561]: s/prefs_path_14/prefs_path_13/ (?) (!)

Untested (yet)

File: src/drivers/Unix/Fl_Unix_System_Driver.cxx [hunk 2]
```diff
@@ -443,87 +444,135 @@ void Fl_Unix_System_Driver::newUUID(char *uuidBuffer)
...
  */
 char *Fl_Unix_System_Driver::preference_rootnode(Fl_Preferences * /*prefs*/,
                                                  Fl_Preferences::Root root,
                                                  const char *vendor,
                                                  const char *application)
 {
-  static char *filename = 0L;
-  if (!filename) filename = (char*)::calloc(1, FL_PATH_MAX);
-  const char *home = "";
+  // Create a static buffer fo our filename
```
Comment (Albrecht-S) [inline src/drivers/Unix/Fl_Unix_System_Driver.cxx:456]: s/fo/for/

File: src/drivers/Unix/Fl_Unix_System_Driver.cxx [hunk 3]
```diff
@@ -443,87 +444,135 @@ void Fl_Unix_System_Driver::newUUID(char *uuidBuffer)
...
-  if ( (application==NULL) || (application[0]==0) )
-    application = "unknown";
+  return prefs_path;
+}
 
-  snprintf(filename + strlen(filename), FL_PATH_MAX - strlen(filename),
-           "%s/%s.prefs", vendor, application);
+/*
+ Memory based preferences are never saved to any file, so the path is
+ just and empty string.
```
Comment (Albrecht-S) [inline src/drivers/Unix/Fl_Unix_System_Driver.cxx:488]: s/and/an/

**Comment by Albrecht-S** [comment]
@MatthiasWM I just reviewed the code (w/o testing), see my comments. AFAICT there was one real bug and the other suggestions fix typos.

Please change the commit message from "Fixes ..." to "Fix ..." ;-)

I'm going to test this ASAP on my system...

**Comment by MatthiasWM** [comment]
Ooops, too fast.

**Comment by Albrecht-S** [comment]
Another issue:
```
[22/35] Building CXX object src/CMakeFiles/fltk.dir/drivers/Unix/Fl_Unix_System_Driver.cxx.o
/git/fltk/master/src/drivers/Unix/Fl_Unix_System_Driver.cxx: In member function ‘char* Fl_Unix_System_Driver::preference_user_rootnode(const char*, const char*, char*)’:
/git/fltk/master/src/drivers/Unix/Fl_Unix_System_Driver.cxx:548:14: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘const int’ [-Wsign-compare]
  548 |     if (h_env!=prefs_path_14.npos)
/git/fltk/master/src/drivers/Unix/Fl_Unix_System_Driver.cxx:551:14: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘const int’ [-Wsign-compare]
  551 |     if (h_env!=prefs_path_14.npos)
```

File: src/drivers/Unix/Fl_Unix_System_Driver.cxx (hunk 1 above)
Comment (MatthiasWM) [reply src/drivers/Unix/Fl_Unix_System_Driver.cxx:561]: Yes, that one would have been an issue!

**Comment by Albrecht-S** [comment]
Looks like everything is fixed now, see also #891.

**Pull request merged**